conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+            description="This key is used to set the OS versions for linux_* platforms. Valid entries map a linux platform and arch to either cos6 or cos7. Currently cos6 is the default for linux-64. All other linux architectures use CentOS 7."
+        ),
+    ] = {p.value: None for p in Platforms if p.value.startswith("linux")}
+
+    provider: Annotated[
+        Union[
+            Dict[Platforms, Union[List[CIservices], CIservices, bool, None]],
+            None,
+        ],
+        Field(
+            description="The provider field is a mapping from build platform (not target platform) to CI service. It determines which service handles each build platform. If a desired build platform is not available with a selected provider (either natively or with emulation), the build will be disabled. Use the build_platform field to manually specify cross-compilation when no providers offer a desired build platform."
+        ),
+    ] = {
+        "linux_64": ["azure"],
+        "osx_64": ["azure"],
+        "win_64": ["azure"],
+        # Following platforms are disabled by default
+        "linux_aarch64": None,
+        "linux_ppc64le": None,
+        "linux_armv7l": None,
+        "linux_s390x": None,
+        # Following platforms are aliases of x86_64,
+        "linux": None,
+        "osx": None,
+        "win": None,
+    }
+
+    recipe_dir: str = Field(
+        description="The relative path to the recipe directory",
+        default="recipe",
+    )
+
+    remote_ci_setup: Annotated[
+        Union[List[str], None],
+        Field(
+            description="This option can be used to override the default conda-forge-ci-setup package. Can be given with ${url or channel_alias}::package_name, defaults to conda-forge channel_alias if no prefix is given."
+        ),
+    ] = ["conda-forge-ci-setup=3"]
+
+    shellcheck: Annotated[
+        Union[ShellCheck, None],
+        Field(
+            description="Shell scripts used for builds or activation scripts can be linted with shellcheck. This option can be used to enable shellcheck and configure its behavior."
+        ),
+    ] = None
+
+    skip_render: Annotated[
+        Union[List[str], None],
+        Field(
+            description="This option specifies a list of files which conda smithy will skip rendering. The possible values can be a subset of .gitignore, .gitattributes, README.md, LICENSE.txt. The default value is an empty list [ ]"
+        ),
+    ] = []
+
+    @field_validator("skip_render")
+    def skip_render_validator(cls, v):
+        if not v:
+            return v
+
+        valid_values = [
+            ".gitignore",
+            ".gitattributes",
+            "README.md",
+            "LICENSE.txt",
+            ".github/workflows",
+        ]
+        for i in v:
+            if i not in valid_values:
+                raise ValueError(
+                    f"Invalid value {i} for skip_render. Valid values are {valid_values}"
+                )
+        return v
```

Comment:
```suggestion
    def skip_render_validator(cls, values):
        if not values:
            return values

        valid_values = [
            ".gitignore",
            ".gitattributes",
            "README.md",
            "LICENSE.txt",
            ".github/workflows",
        ]
        for value in values:
            if value not in valid_values:
                raise ValueError(
                    f"Invalid value '{value}' for skip_render. Valid values are {valid_values}"
                )
        return values
```

Although I think this can be more easily computed with a set difference or something like that.